P2PEnabledConference.js

```diff
@@ -0,0 +1,947 @@
+
+        logger.info(`Adding ${localTracks} to P2P...`);
+        this.p2pJingleSession.addLocalTracks(localTracks).then(
+            () => {
+                logger.info(`Added ${localTracks} to P2P`);
+                logger.info('About to send P2P \'session-initiate\'...');
+                this.p2pJingleSession.invite();
+            },
+            error => {
+                logger.error(`Failed to add ${localTracks} to P2P`, error);
+            });
+    }
+
+    /**
+     * Method when called will decide whether it's the time to start or stop the
+     * P2P session.
+     * @param {boolean} userLeftEvent if <tt>true</tt> it means that the call
+     * originates from the user left event.
+     * @private
+     */
+    _startStopP2PSession(userLeftEvent) {
+        if (this.options.config.disableAutoP2P) {
+            logger.info('Auto P2P disabled');
+
+            return;
+        }
+        const peers = this.getParticipants();
+        const peerCount = peers.length;
+        const isModerator = this.isModerator();
+
+        // FIXME 1 peer and it must *support* P2P switching
+        const shouldBeInP2P = peerCount === 1;
+
+        logger.debug(
+            `P2P? isModerator: ${isModerator
+             }, peerCount: ${peerCount} => ${shouldBeInP2P}`);
+
+        // Clear deferred "start P2P" task
+        if (!shouldBeInP2P && this.deferredStartP2P) {
+            logger.info('Cleared deferred start P2P task');
+            window.clearTimeout(this.deferredStartP2P);
+            this.deferredStartP2P = null;
+        }
+
+        // Start peer to peer session
+        if (isModerator && !this.p2pJingleSession && shouldBeInP2P) {
+            const peer = peerCount && peers[0];
+
+            // Everyone is a moderator ?
+            if (isModerator && peer.getRole() === 'moderator') {
+                const myId = this.myUserId();
+                const peersId = peer.getId();
+
+                if (myId > peersId) {
+                    logger.debug(
+                        'Everyone\'s a moderator - '
+                            + 'the other peer should start P2P', myId, peersId);
+
+                    return;
+                } else if (myId === peersId) {
+                    logger.error('The same IDs ? ', myId, peersId);
+
+                    return;
+                }
+            }
+            const jid = peer.getJid();
+
+            if (userLeftEvent) {
+                if (this.deferredStartP2P) {
+                    logger.error(
+                        'Deferred start P2P task\'s been set already!');
+
+                    return;
+                }
+                logger.info(
+                    `Will start P2P with: ${jid
+                         } after ${this.backToP2PDelay} seconds...`);
+                this.deferredStartP2P = window.setTimeout(
+                    this._startPeer2PeerSession.bind(this, jid),
+                    this.backToP2PDelay * 1000);
```

Comment:
well this way it can be stored in seconds which is more human friendly in my opinion